ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLProvider.java

```diff
@@ -74,4 +82,18 @@ private static RuntimeWiring buildWiring(final GraphQLDataFetchers graphQLDataFe
                     "sendRawTransaction", graphQLDataFetchers.getSendRawTransactionDataFetcher()))
         .build();
   }
+
+  private static int calculateFieldCost(
+      final FieldComplexityEnvironment environment, final int childComplexity) {
+    final String childTypeName = environment.getParentType().getName();
+    final String fieldName = environment.getField().getName();
+
+    if (childTypeName.equals("Transaction") && fieldName.equals("block")) {
+      return childComplexity + 100;
+    } else if (childTypeName.equals("__Type") && fieldName.equals("fields")) {
+      return childComplexity + 100;
+    } else {
+      return childComplexity + 1;
```

Comment:
Is there a published algorithm for this?

Reply:
This is a custom algorithm, focused on our schema.